helix-core/src/main/java/org/apache/helix/manager/zk/ZKUtil.java

```diff
@@ -42,6 +43,14 @@
   private ZKUtil() {
   }
 
+  public static boolean isClusterSetup(String clusterName, String zkAddress) {
+    HelixZkClient zkClient = getHelixZkClient(zkAddress);
+    boolean result = isClusterSetup(clusterName, zkClient);
+    zkClient.close();
```

Comment:
Just a throught, shall we have a static ZKClient map instead of creating/closing for each request? We can close all of them with a close() function? Or setup a timeout for idle usage?

Reply:
Another approach is to have a static dedicated ZK client. But I made the design choice to create a ZK client and closing at the end of the method for the following reasons:
1. ZKUtil's methods are static and therefore should be stateless.
2. The usages for ZKUtil should generally be a one-time thing.
3. It is not desirable to have a static connection running at all times; you'd never know when the connection should be closed.

This is a tradeoff I made considering the points above. Of course, the drawback of this approach is that creating a dedicated connection for each static method call is not a lightweight operation.

What do people think? @lei-xia @jiajunwang ?

Reply:
I buy the idea of one time access of different ZKs. But I think we should do following things:
1. Make clear Java doc to let user understand it could cause more resource usage (ZKClient open/close).
2. We should not deprecate the old API as long as that can provide user an option to now open/close ZKClient frequently while they would like to use the similar functionalities.

Also we need to clearly define the role of this Util and follow the rule for future development.

Reply:
@dasahcc 

Those are reasonable points. What I'll do is that 
1. I will remove @deprecated annotation from existing methods.
2. I will add in the Javadoc for newly added methods saying that it will create a ZK connection every time. 
3. In general, ZKUtil should be used for testing and Helix internal uses only. I will clarify this in the Javadoc for the class.

Reply:
> I buy the idea of one time access of different ZKs. But I think we should do following things:
> 
> 1. Make clear Java doc to let user understand it could cause more resource usage (ZKClient open/close).
> 2. We should not deprecate the old API as long as that can provide user an option to now open/close ZKClient frequently while they would like to use the similar functionalities.
> 
> Also we need to clearly define the role of this Util and follow the rule for future development.

@dasahcc, actually we are targeting to hide ZkClient from Helix powered applications. So that later they don't see them anymore. Instead, they see HelixManager, HelixAdmin, ConfigAccessor, HelixDataAccessor, HelixPropertyStore, HelixVerifiyer and the last one is ZkUtil.

Reply:
@dasahcc Your points have been addressed.